hibernate-core/src/main/java/org/hibernate/metamodel/internal/MetamodelImpl.java

```diff
@@ -623,12 +625,20 @@ else if ( result == INVALID_IMPORT ) {
 	 */
 	public String[] getImplementors(String className) throws MappingException {
 
+		if (implementorsCache.containsKey(className)) {
+			return implementorsCache.get(className);
```

Comment:
Better call `get` then test if the result is not null, that's one less map access.